plugins/item_tasks/server/__init__.py

```diff
@@ -40,8 +41,8 @@ def _onUpload(event):
     they are used to link item task outputs back to a job document.
     """
     try:
-        ref = json.loads(event.info.get('reference'))
-    except (ValueError, TypeError):
+        ref = json.loads(event.info.get('reference', ''))
+    except json.JSONDecodeError if sys.version_info >= (3, 5) else ValueError:
```

Comment:
What happened to the `TypeError` catching?

Reply:
If it exists, `'reference'` will always be a string (since [it came from the HTTP request](https://github.com/girder/girder/blob/master/girder/models/upload.py#L254)). The only case a `TypeError` should be emitted would be if the key was not found, and `None` was returned by default. By returning `''` by default instead, a `TypeError` *should* no longer occur.

Indeed, if a `TypeError` does occur in the future, it means that one of our preconditions around the `'reference'` data type was violated, so we don't want to blindly suppress the `TypeError`.